code/mission/missionparse.cpp

```diff
@@ -6068,10 +6071,21 @@ void mission_set_wing_arrival_location( wing *wingp, int num_to_set )
 		}
 	}
 
-	// create warp effect if in mission and not arriving from docking bay
-	if ( (Game_mode & GM_IN_MISSION) && (wingp->arrival_location != ARRIVE_FROM_DOCK_BAY) ) {
+	if (Game_mode & GM_IN_MISSION) {
 		for ( index = wingp->current_count - num_to_set; index < wingp->current_count; index ++ ) {
-			shipfx_warpin_start( &Objects[Ships[wingp->ship_index[index]].objnum] );
+			object *objp = &Objects[Ships[wingp->ship_index[index]].objnum];
+
+			if (anchor_objnum >= 0)
+				Script_system.SetHookObjects(2, "Ship", objp, "Parent", &Objects[anchor_objnum]);
+			else
+				Script_system.SetHookObjects(2, "Ship", objp, "Parent", NULL);
+
+			Script_system.RunCondition(CHA_ONSHIPARRIVE, 0, NULL, objp);
+			Script_system.RemHookVars(2, "Ship", "Parent");
+
+			if (wingp->arrival_location != ARRIVE_FROM_DOCK_BAY) {
```

Comment:
I removed it because it was rather pointless; the code is pretty much as clear and self-documenting as it possibly can be WRT the fact that the warp effect is only created "if in mission and not arriving from docking bay". Plus the "in mission" conditional was moved further up, so that part of the comment would have needed updating anyway.

So, I figured that having a comment+code combo like this would be borderline silly:

```
// create warp effect if not arriving from docking bay
if (wingp->arrival_location != ARRIVE_FROM_DOCK_BAY) {
	shipfx_warpin_start(objp);
}
```